Add support for scripting tensor.type

nikithamalgifb opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51903 Added support in builtin_functions.cpp

Fixes issue: #25433

Summary:
=========
Makes tensor.type(dtype) scriptable

Test:
======
python test/test_jit.py -v TestJit.test_script_tensor_type

Commit:
Add support for scripting tensor.type

[ghstack-poisoned]

Commit:
Update on "Add support for scripting tensor.type"


Fixes issue: #25433

Summary:
=========
Makes tensor.type(dtype) scriptable

Test:
======
python test/test_jit.py -v TestJit.test_script_tensor_type

[ghstack-poisoned]

eellison commented on torch/csrc/jit/runtime/register_prim_ops.cpp:
```diff
@@ -987,6 +987,22 @@ RegisterOperators reg(
          aliasAnalysisFromSchema()),
+     OperatorGenerator(
+         TORCH_SELECTIVE_SCHEMA(
+             "aten::type.prim_dtype(Tensor(a) self, int? dtype=None, bool non_blocking=False, bool copy=False) -> Tensor(a|b)"),
```
This is the exact same schema as `aten::to.prim_dtype(Tensor(a) self, int? dtype=None, bool non_blocking=False, bool copy=False`, and does the same thing. I think it would be simpler to add it by adding `(torch.type, "aten::to")` to the builtin operator registry here

nikithamalgifb replied:
I agree. I tried looking earlier if I could just connect `type` -> `to` but couldn't find such an example.
Do you mean like this?
```
OperatorGenerator(torch.type, "aten::to"),
```

eellison reviewed (commented):
Looks good, I think it would be simpler and less code if we reused `aten::to` here instead of introducing a new operator.

Commit:
Update on "Add support for scripting tensor.type"


Fixes issue: #25433

Summary:
=========
Makes tensor.type(dtype) scriptable

Test:
======
python test/test_jit.py -v TestJit.test_script_tensor_type

[ghstack-poisoned]

nikithamalgifb commented:
Messed up the PR with ghstack. Closing this PR
